python/pycrdt/_awareness.py

```diff
@@ -217,43 +199,3 @@ def unobserve(self, id: str) -> None:
             id: The subscription ID to unregister.
         """
         del self._subscriptions[id]
-
-    def _update_states(
-        self, client_id: int, clock: int, state: Any, states_changes: dict[str, list[int]]
-    ) -> None:
```

Comment:
I think this function could be kept, using an `origin` to know if we should remove the local state or not.

Reply:
But what about [this](https://github.com/yjs/y-protocols/blob/255b7e6c96d9c6dd48b5519b3f83ab65afdb4bba/awareness.js#L256-L284), that `_update_states()` was doing inconditionally?

Reply:
I agree that it was wrong in `_update_states()`, it should also have tested that origin is not "local".
Anyway, the current implementation works, it was just to avoid duplicating code.